In [1]:
import sys
sys.path.append("../")
sys.path.append('C:\Users\James Pino\PycharmProjects\Magine')
from IPython.display import display
%matplotlib inline
import matplotlib.pyplot as plt
import pandas as pd
import networkx as nx
import magine.ontology.enrichment_tools as et
import magine.networks.visualization.notebook_tools as nt
import magine.networks.visualization.notebooks.view as view
from magine.networks.network_subgraphs import NetworkSubgraphs
from exp_data import exp_data

<IPython.core.display.Javascript object>

# Exploring network using enrichment analysis

### This example uses the enrichment output we just obtained to explore the network.

First, lets load the molecular network and the enrichment output from the previous runs. 

In [2]:
enrichment_array = pd.read_csv('Data/cisplatin_enrichment.csv.gz', index_col=0)
network = nx.read_gpickle('Data/cisplatin_based_network.p')
subgraph_gen = NetworkSubgraphs(network=network, exp_data=exp_data)

For this example, we will only look at the proteomics data. 

In [3]:
proteomics = enrichment_array[enrichment_array['category'].str.contains('proteomics')].copy()

For this example, we just want to look at "biological processes" descriptions, so we will limit our analysis to only databases with this type of information

In [4]:
print(proteomics['db'].unique())

process_dbs = [
        'GO_Biological_Process_2017',
        'Humancyc_2016',
        'Reactome_2016',
        'KEGG_2016',
        'NCI-Nature_2016',
        'Panther_2016',
        'WikiPathways_2016',
]

display(proteomics.head(10))

['KEGG_2016' 'NCI-Nature_2016' 'Panther_2016' 'WikiPathways_2016'
 'BioCarta_2016' 'Humancyc_2016' 'Reactome_2016' 'KEA_2015'
 'GO_Biological_Process_2017' 'GO_Molecular_Function_2017'
 'GO_Cellular_Component_2017' 'ChEA_2016' 'TRANSFAC_and_JASPAR_PWMs'
 'ENCODE_TF_ChIP-seq_2015' 'Kinase_Perturbations_from_GEO_down'
 'Kinase_Perturbations_from_GEO_up'
 'LINCS_L1000_Kinase_Perturbations_down'
 'LINCS_L1000_Kinase_Perturbations_up'
 'Ligand_Perturbations_from_GEO_down' 'Ligand_Perturbations_from_GEO_up'
 'DrugMatrix' 'Drug_Perturbations_from_GEO_2014']


term_name  rank  \
0  Spliceosome_Homo sapiens_hsa03040                             1      
1  Ribosome_Homo sapiens_hsa03010                                2      
2  Salmonella infection_Homo sapiens_hsa05132                    3      
3  Bacterial invasion of epithelial cells_Homo sapiens_hsa05100  4      
4  Ubiquitin mediated proteolysis_Homo sapiens_hsa04120          5      
5  HTLV-I infection_Homo sapiens_hsa05166                        6      
6  Pathogenic Escherichia coli infection_Homo sapiens_hsa05130   7      
7  RNA transport_Homo sapiens_hsa03013                           8      
8  Huntington's disease_Homo sapiens_hsa05016                    9      
9  mRNA surveillance pathway_Homo sapiens_hsa03015               10     

   combined_score   adj_p_value  \
0  90.501003       3.360688e-20   
1  55.241281       1.476445e-12   
2  16.060192       5.467710e-03   
3  12.601933       2.643585e-02   
4  12.002185       2.643585e-02   
5  11.920403       4.686993e-02   
6  11.801291       3.381820e-02   
7  10.912681       4.799956e-02   
8  10.478332       4.799956e-02   
9  9.547068        4.799956e-02   

                                                                                                                                                                                                                                       genes  \
0  ALYREF,DDX39B,DDX5,HNRNPA1,HNRNPA3,HNRNPC,HNRNPM,HNRNPU,HSPA2,PRPF19,PRPF3,PRPF8,PUF60,RBM8A,RBMX,SF3A1,SF3B1,SNRNP70,SNRPA,SNRPB,SNRPC,SNRPD1,SNRPD2,SNRPD3,SNRPE,SNRPF,SNW1,SRSF1,SRSF3,SRSF4,SRSF5,SRSF6,SRSF7,SRSF9,THOC2,U2AF1,U2AF2   
1  RPL11,RPL13,RPL13A,RPL14,RPL15,RPL17,RPL18A,RPL19,RPL21,RPL22,RPL23A,RPL26L1,RPL29,RPL3,RPL36,RPL37A,RPL5,RPL7,RPL7A,RPL8,RPLP0,RPLP1,RPS14,RPS18,RPS19,RPS26,RPS3,RPS5,RPS6                                                                
2  ARPC1A,ARPC4,DYNC1H1,DYNC2H1,FLNB,KLC2,MAPK3,MYH10,MYH9,NLRC4,RELA,ROCK2                                                                                                                                                                    
3  ARPC1A,ARPC4,CAV1,CD2AP,CLTCL1,CTNNB1,CTTN,MET,SEPT9,SHC1                                                                                                                                                                                   
4  BRCA1,CDC16,CDC20,CUL3,CUL4A,CUL4B,HERC4,HUWE1,PIAS1,PRPF19,SAE1,TCEB1,UBA1,UBE4B                                                                                                                                                           
5  APC,CALR,CDC16,CDC20,CHEK1,CTNNB1,HLA-DQA2,IKBKG,MAP2K4,MAP3K3,MYB,MYC,RANBP3,RELA,RRAS2,SLC25A5,SLC25A6,TLN1,VDAC1,VDAC3                                                                                                                   
6  ARPC1A,ARPC4,CTNNB1,CTTN,KRT18,NCL,ROCK2,YWHAZ                                                                                                                                                                                              
7  ALYREF,DDX39B,EIF2S2,EIF2S3,EIF3B,EIF4EBP2,NUP188,NUP88,NUP98,PABPC1,PABPC3,PABPC4,RBM8A,THOC2,TPR                                                                                                                                          
8  CLTCL1,COX4I1,COX5A,COX5B,COX6C,DCTN1,DNAH17,HDAC1,NDUFB9,SLC25A5,SLC25A6,TGM2,UQCRC1,UQCRC2,VDAC1,VDAC3                                                                                                                                    
9  ALYREF,CPSF6,DAZAP1,DDX39B,FIP1L1,PABPC1,PABPC3,PABPC4,PPP1CC,RBM8A                                                                                                                                                                         

   n_genes sample_id         category         db  
0  37       01hr      proteomics_both  KEGG_2016  
1  29       01hr      proteomics_both  KEGG_2016  
2  12       01hr      proteomics_both  KEGG_2016  
3  10       01hr   

In [5]:
time_1_hour_prot = et.filter_dataframe(proteomics, 
                                       p_value=0.05,
                                       combined_score=0.0, 
                                       category=['proteomics_up'], 
                                       sample_id='01hr', 
                                       db=process_dbs
                                      )
display(time_1_hour_prot.head(10))

term_name  rank  \
189887  negative regulation of transcription, DNA-templated              1      
189000  Cell Cycle_Homo sapiens_R-HSA-1640170                            1      
189001  Interleukin-2 signaling_Homo sapiens_R-HSA-451927                2      
189889  resolution of meiotic recombination intermediates                3      
189002  Interleukin-3, 5 and GM-CSF signaling_Homo sapiens_R-HSA-512988  3      
189003  Cell Cycle, Mitotic_Homo sapiens_R-HSA-69278                     4      
189004  Interleukin receptor SHC signaling_Homo sapiens_R-HSA-912526     5      
189005  MAPK family signaling cascades_Homo sapiens_R-HSA-5683057        6      
189006  Signaling by FGFR2_Homo sapiens_R-HSA-5654738                    7      
189007  Insulin receptor signalling cascade_Homo sapiens_R-HSA-74751     8      

        combined_score  adj_p_value  \
189887  68.261978       0.001601      
189000  36.507320       0.000267      
189001  30.220851       0.001586      
189889  29.557281       0.016485      
189002  29.303950       0.001586      
189003  27.727089       0.001586      
189004  27.242088       0.001586      
189005  26.440058       0.001586      
189006  25.922377       0.001586      
189007  25.791980       0.001586      

                                                                                                                                                                genes  \
189887  ARID5B,BCLAF1,BRCA1,CALR,CBY1,CHD8,ENO1,HDAC1,ILF3,KHDRBS1,LIMS1,LRRFIP1,MYB,RBM15,RSF1,SHC1,SPEN,XRCC6,ZBED6,ZC3H8,ZNF350,ZNF552                               
189000  ACD,AKAP9,BRCA1,CDC16,CDC20,CDC7,CLASP2,DCTN1,HAUS8,HDAC1,HSPA2,KNTC1,LMNA,MAPK3,MYC,NPM1,NUP188,NUP88,PDS5B,PLK1,PRIM1,RBL1,RSF1,SFI1,SMC1A,TOP2A,TYMS,YWHAZ   
189001  AKAP9,BRAF,CNKSR2,CUL3,HAVCR2,INPPL1,IRS2,MAPK3,MARK3,PTPRA,RASGRF2,SHC1,SPTBN1,SPTBN2,TLN1,VWF                                                                 
189889  ERCC4,SLX4,TOP2A,TOP2B                                                                                                                                          
189002  AKAP9,BRAF,CNKSR2,CUL3,INPPL1,IRS2,MAPK3,MARK3,PTPRA,RASGRF2,SHC1,SPTBN1,SPTBN2,TLN1,VWF,YWHAZ                                                                  
189003  AKAP9,CDC16,CDC20,CDC7,CLASP2,DCTN1,HAUS8,HDAC1,KNTC1,LMNA,MAPK3,MYC,NUP188,NUP88,PDS5B,PLK1,PRIM1,RBL1,SFI1,SMC1A,TOP2A,TYMS                                   
189004  AKAP9,BRAF,CNKSR2,CUL3,INPPL1,IRS2,MAPK3,MARK3,PTPRA,RASGRF2,SHC1,SPTBN1,SPTBN2,TLN1,VWF                                                                        
189005  AKAP9,BRAF,CNKSR2,CUL3,DNAJB1,IRS2,MAPK3,MARK3,MYC,PTPRA,RASGRF2,SHC1,SPTBN1,SPTBN2,TLN1,VWF                                                                    
189006  AKAP9,BRAF,CNKSR2,CUL3,HNRNPA1,HNRNPM,INSR,IRS2,MAPK3,MAPKAP1,MARK3,PTPRA,RASGRF2,SHC1,SPTBN1,SPTBN2,TLN1,VWF                                                   
189007  AKAP9,BRAF,CNKSR2,CUL3,INSR,IRS2,MAPK3,MARK3,PIK3R4,PTPRA,RASGRF2,SHC1,SPTBN1,SPTBN2,TLN1,VWF                                                                   

        n_genes sample_id       category                          db  
189887  22       01hr      proteomics_up  GO_Biological_Process_2017  
189000  28       01hr      proteomics_up  Reactome_2016               
189001  16       01hr      proteomics_up  Reactome_2016               
189889  4        01hr      proteomics_up  GO_Biological_Process_2017  
189002  16       01hr      proteomics_up  Reactome_2016               
189003  22       01hr      proteomics_up  Reactome_2016               
189004  15       01hr      proteomics_up  Reactome_2016               
189005  16       01hr      proteomics_up  Reactome_2016               
189006  18       01hr      proteomics_up  Reactome_2016               
189007  16       01hr      proteomics_up  Reactome_2016

Since terms across databases might be redundant ("Interleukin-3, 5 and GM-CSF signaling_Homo sapiens_R-HSA-512988" and "Interleukin receptor SHC signaling_Homo sapiens_R-HSA-912526" have nearly a full overlap of genes), we want to eliminate duplicate terms and focus on the most enriched. For that we use the Jaccard Index. It is implmented in the find_similar_terms function in enrichment_tools (et).

In [6]:
filtered_1hr = et.filter_similar_terms(time_1_hour_prot, threshold=.7)
display(filtered_1hr)

Number of rows went from 89 to 25


term_name  \
189887  negative regulation of transcription, DNA-templated                                                       
189000  Cell Cycle_Homo sapiens_R-HSA-1640170                                                                     
189001  Interleukin-2 signaling_Homo sapiens_R-HSA-451927                                                         
189889  resolution of meiotic recombination intermediates                                                         
189008  Signalling by NGF_Homo sapiens_R-HSA-166520                                                               
189893  mitotic cytokinesis                                                                                       
188581  XPodNet - protein-protein interactions in the podocyte expanded by STRING_Mus musculus_WP2309             
189894  chloride transport                                                                                        
189032  Axon guidance_Homo sapiens_R-HSA-422475                                                                   
188582  Retinoblastoma (RB) in Cancer_Homo sapiens_WP2446                                                         
188583  EGFR1 Signaling Pathway_Mus musculus_WP572                                                                
188379  Validated targets of C-MYC transcriptional repression_Homo sapiens_6bbdafa6-6196-11e5-8ac5-06603eb7f303   
188380  Regulation of Telomerase_Homo sapiens_4dfe97ca-6195-11e5-8ac5-06603eb7f303                                
188584  Cell Cycle_Homo sapiens_WP179                                                                             
189056  Deactivation of the beta-catenin transactivating complex_Homo sapiens_R-HSA-3769402                       
188381  E2F transcription factor network_Homo sapiens_bb4d0fd3-6191-11e5-8ac5-06603eb7f303                        
189059  Immune System_Homo sapiens_R-HSA-168256                                                                   
188382  Notch signaling pathway_Homo sapiens_88f83518-6194-11e5-8ac5-06603eb7f303                                 
189062  SUMO E3 ligases SUMOylate target proteins_Homo sapiens_R-HSA-3108232                                      
189063  DNA Repair_Homo sapiens_R-HSA-73894                                                                       
189067  Signaling by Rho GTPases_Homo sapiens_R-HSA-194315                                                        
189071  Unfolded Protein Response (UPR)_Homo sapiens_R-HSA-381119                                                 
189072  Processing of Capped Intron-Containing Pre-mRNA_Homo sapiens_R-HSA-72203                                  
189085  Platelet Aggregation (Plug Formation)_Homo sapiens_R-HSA-76009                                            
189087  Apoptotic cleavage of cellular proteins_Homo sapiens_R-HSA-111465                                         

        rank  combined_score  adj_p_value  \
189887  1     68.261978       0.001601      
189000  1     36.507320       0.000267      
189001  2     30.220851       0.001586      
189889  3     29.557281       0.016485      
189008  9     25.783068       0.001586      
189893  7     24.798241       0.030953      
188581  1     22.460249       0.011088      
189894  8     22.310453       0.029255      
189032  33    20.509597       0.001586      
188582  2     17.166900       0.017626      
188583  3     15.618928       0.041761      
188379  1     15.369523       0.009573      
188380  2     14.719095       0.009573      
188584  4     14.706942       0.029875      
189056  57    14.531321       0.001586      
188381  3     14.331688       0.010065      
189059  60    12.588445       0.008988      
188382  4     11.237926       0.016905      
189062  63    9.932085        0.013351      
189063  64    9.569204        0.032067      
189067  68    8.899833        0.036040      
189071  72    8.076945        0.032067      
189072  73    7.805714        0.048395      
189085  86    5.412894        0

Now we can explore the top hits, which has been slimmed from 89 to 33 terms. Generally this is where the expert knowledge comes in. However, a quick search with each term and search terms of you molecule of interest tend to be useful. 

The first hit is 'negative regulation of transcription', which means that something caused genes not to be transcribed. Cisplatin causes DNA damage, thus negative regulaton of transcription makes sense. So does top hit 2, 'Cell Cycle_Homo sapiens_R-HSA-1640170'. 

A quick search for 'Interleukin-2 signaling' and 'Cisplatin' __[link](https://www.google.com/search?rlz=1C1CHBD_enUS721US721&ei=KzNeWuCxBsfq_AaSgYuYDQ&q=Interleukin-2+signaling+cisplatin&oq=Interleukin-2+signaling+cisplatin&gs_l=psy-ab.3..35i39k1.8097.9052.0.9196.10.10.0.0.0.0.145.897.7j3.10.0....0...1c.1.64.psy-ab..3.2.218....0.TInUjcZY740)__ returns a paper titled "Cisplatin at clinically relevant concentrations enhances interleukin-2 synthesis by human primary blood lymphocytes." __[link](https://www.ncbi.nlm.nih.gov/pubmed/10211553)__


We can link the two together to see how once might regulate the other by looking at the molecular interactions.

## Expected findings

In [7]:
shorten_names = {
    'Cell Cycle_Homo sapiens_R-HSA-1640170':'Cell Cycle',
    'DNA Repair_Homo sapiens_R-HSA-73894' : 'DNA Repair',
    'Interleukin-2 signaling_Homo sapiens_R-HSA-451927': 'IL2',
    'resolution of meiotic recombination intermediates': 'Meiotic Recombination'
                }
renamed_1hr = filtered_1hr.copy()
renamed_1hr['term_name'] = renamed_1hr['term_name'].replace(shorten_names)
term_net_1, mol_net_1 = nt.create_subnetwork(shorten_names.values(), renamed_1hr, network, '1hr')

('DNA Repair', 12)
('IL2', 16)
('Meiotic Recombination', 4)
('Cell Cycle', 28)
Looking for direct edges
No edges between DNA Repair and IL2
No edges between Meiotic Recombination and IL2


In [8]:
dna_genes = et.term_to_genes(df=renamed_1hr, term='DNA Repair')
dna_network = subgraph_gen.shortest_paths_between_lists(dna_genes, max_length=3)
print(dna_network.nodes())

Removing from list
['BRCA1', u'UBE2I', 'XRCC6', 'SLX4', 'UVSSA', 'ERCC4', 'CUL4A', 'XRCC1', 'PIAS1', u'DDB2', u'MDM2', u'EP300', 'CUL4B']


In [ ]:
subgraph_gen.measured_networks_over_time(dna_network, ['red', 'green', 'blue', 'purple'], 'dna')


In [9]:
view.render_graph(dna_network)

In [10]:
view.display_graph(term_net_1)

In [11]:
view.render_graph(mol_net_1, add_parent=False)

## Exploring other top hits from 1 hour

### Side effects of cisplatin
Chemotherapy-induced peripheral neuropathy. __[link](https://www.frontiersin.org/articles/10.3389/fnins.2017.00481/full)__
It is not well understood why cisplatin causes CIPN. Surprisely here, we see that Axon Guidance has a combined score of 20.5. 33 species are effected by cisplain that are linked with axon guidance. We are not neural experts and did not know that axon guidance was related to CIPN. Using MAGINE we were able to find ties between the two.

In [12]:
axon_guidance = et.term_to_genes(df=renamed_1hr, term='Axon guidance_Homo sapiens_R-HSA-422475')
print(axon_guidance)

set(['NRCAM', 'SRGAP2', 'CNKSR2', 'SPTBN2', 'SPTBN1', 'SHC1', 'TLN1', 'MAPK3', 'MARK3', 'CRMP1', 'CUL3', 'CLASP2', 'CLTCL1', 'SDCBP', 'RASGRF2', 'ARPC1A', 'AKAP9', 'IRS2', 'MET', 'BRAF', 'PTPRA', 'VWF'])


In [13]:
g = subgraph_gen.neighbors_of_list(axon_guidance, up_stream=True, down_stream=False, max_dist=1, 
                                   include_only=exp_data.sig_species_over_time['01hr']
                                  )
nt.render_graph(g)

In [15]:
chloride_transport = et.term_to_genes(df=renamed_1hr, term='chloride transport')
print(chloride_transport)
chl_trans = subgraph_gen.neighbors_of_list(chloride_transport, max_dist=2, 
#                                            include_only=exp_data.list_species
                                          )
nt.display_graph(chl_trans)

set(['ANO4', 'ANO5', 'ANO6', 'SLC26A8', 'CLIC4'])


# 6 hour time point

In [16]:
time_6_hour_prot = et.filter_dataframe(proteomics, p_value=0.05, combined_score=0.0, category='proteomics_up', sample_id='06hr', db=process_dbs)
slimmed = et.filter_similar_terms(time_6_hour_prot)
display(slimmed.head(25))

Number of rows went from 265 to 141


term_name  \
211872  mRNA splicing, via spliceosome                                                                  
210292  mRNA processing_Mus musculus_WP310                                                              
211873  positive regulation of transcription from RNA polymerase II promoter                            
210798  Gene Expression_Homo sapiens_R-HSA-74160                                                        
211874  RNA splicing                                                                                    
211875  mRNA export from nucleus                                                                        
211876  rRNA processing                                                                                 
211877  protein sumoylation                                                                             
209818  RNA transport_Homo sapiens_hsa03013                                                             
209819  Spliceosome_Homo sapiens_hsa03040                                                               
210293  XPodNet - protein-protein interactions in the podocyte expanded by STRING_Mus musculus_WP2309   
211878  cellular response to DNA damage stimulus                                                        
210800  Activation of the AP-1 family of transcription factors_Homo sapiens_R-HSA-450341                
210294  EGFR1 Signaling Pathway_Mus musculus_WP572                                                      
211879  RNA processing                                                                                  
210295  mRNA Processing_Homo sapiens_WP411                                                              
210802  SUMOylation_Homo sapiens_R-HSA-2990846                                                          
211880  regulation of cellular response to heat                                                         
210805  Cell Cycle_Homo sapiens_R-HSA-1640170                                                           
210296  TSLP Signaling Pathway_Homo sapiens_WP2203                                                      
211882  response to cAMP                                                                                
210297  TGF-beta Signaling Pathway_Homo sapiens_WP366                                                   
211883  nuclear-transcribed mRNA catabolic process, nonsense-mediated decay                             
210808  Growth hormone receptor signaling_Homo sapiens_R-HSA-982772                                     
211884  osteoblast differentiation                                                                      

        rank  combined_score   adj_p_value  \
211872  1     87.396296       1.684701e-07   
210292  1     57.331009       6.868228e-10   
211873  2     56.402442       6.010178e-03   
210798  1     55.510780       5.921719e-09   
211874  3     50.639384       1.510552e-05   
211875  4     46.486449       2.371966e-05   
211876  5     45.422599       3.304134e-04   
211877  6     42.216525       6.080538e-05   
209818  1     41.821069       4.495322e-08   
209819  2     38.345407       4.797907e-08   
210293  2     36.008199       6.722057e-06   
211878  7     35.101473       2.392831e-03   
210800  3     33.437382       1.176482e-05   
210294  3     33.347966       1.160157e-05   
211879  8     32.992801       3.304134e-04   
210295  4     32.884630       6.722057e-06   
210802  5     31.744151       1.176482e-05   
211880  9     31.308147       6.655661e-04   
210805  8     29.160810       1.764157e-04   
210296  5     29.017371       2.093567e-05   
211882  11    28.208944       5.029015e-04   
210297  6     27.518886       2.270499e-05   
211883  12    26.712485       9.010199e-03   
210808  11    26.482133       1.508107e-04   
211884  13    26.038610       5.052837e-03   

                                                                                                                                                                    

In [17]:
shorten_names = {
    'protein sumoylation':'protein sumoylation',
    'Activation of the AP-1 family of transcription factors_Homo sapiens_R-HSA-450341' : 'AP1 activation',
    'response to cAMP' : 'response to cAMP',
                }
time_6_hour_prot['term_name'] = time_6_hour_prot['term_name'].replace(shorten_names)
term_net_6, mol_net_6 = nt.create_subnetwork(shorten_names.values(), time_6_hour_prot, network, '06hr', cytoscape_js=False)


('response to cAMP', 5)
('protein sumoylation', 12)
('AP1 activation', 6)
Looking for direct edges


In [18]:
view.display_graph(term_net_6)

In [19]:
view.render_graph(mol_net_6)

In [20]:
# nt.find_neighbors(g=mol_net_1, start='ATR', up_stream=False, down_stream=True, max_dist=1, render=True)
# nt.find_neighbors(g=mol_net_1, start='ATR', up_stream=True, down_stream=False, max_dist=3, render=True)

In [21]:
time_24_hour_prot = et.filter_dataframe(proteomics, p_value=0.05, combined_score=0.0, category='proteomics_up', sample_id='24hr', db=process_dbs)
slimmed = et.filter_similar_terms(time_24_hour_prot, threshold=.5)
display(slimmed.head(25))

Number of rows went from 378 to 141


term_name  \
234775  mRNA splicing, via spliceosome                                               
233501  Gene Expression_Homo sapiens_R-HSA-74160                                     
234776  rRNA processing                                                              
234777  regulation of cellular response to heat                                      
232930  mRNA processing_Mus musculus_WP310                                           
234778  mRNA export from nucleus                                                     
233504  Cell Cycle_Homo sapiens_R-HSA-1640170                                        
234780  positive regulation of apoptotic process                                     
234781  RNA splicing                                                                 
234783  regulation of alternative mRNA splicing, via spliceosome                     
233514  Infectious disease_Homo sapiens_R-HSA-5663205                                
233518  SUMOylation_Homo sapiens_R-HSA-2990846                                       
232407  RNA transport_Homo sapiens_hsa03013                                          
234785  mRNA processing                                                              
233524  Apoptosis_Homo sapiens_R-HSA-109581                                          
232932  Leptin signaling pathway_Homo sapiens_WP2034                                 
232933  Integrated Breast Cancer Pathway_Homo sapiens_WP1984                         
232934  EGF/EGFR Signaling Pathway_Homo sapiens_WP437                                
232935  EGFR1 Signaling Pathway_Mus musculus_WP572                                   
232657  Regulation of Telomerase_Homo sapiens_4dfe97ca-6195-11e5-8ac5-06603eb7f303   
234793  neutrophil degranulation                                                     
234794  cellular response to DNA damage stimulus                                     
234796  positive regulation of protein export from nucleus                           
234797  negative regulation of apoptotic process                                     
232936  Integrated Pancreatic Cancer Pathway_Homo sapiens_WP2377                     

        rank  combined_score   adj_p_value  \
234775  1     209.979563      3.357108e-21   
233501  2     102.139686      1.220822e-18   
234776  2     96.343182       4.915441e-10   
234777  3     84.291259       2.769570e-11   
232930  1     75.595732       1.401438e-13   
234778  4     73.912638       1.055205e-09   
233504  5     62.048920       6.097046e-10   
234780  6     53.817781       5.123684e-05   
234781  7     50.724093       2.169909e-06   
234783  9     45.264258       4.438842e-07   
233514  15    43.423305       1.147292e-07   
233518  19    41.460201       1.127330e-08   
232407  2     38.738445       9.324305e-08   
234785  11    38.279921       1.025094e-05   
233524  25    36.840898       1.683367e-07   
232932  3     34.391387       7.235247e-06   
232933  4     32.691245       9.264562e-06   
232934  5     32.542271       7.235247e-06   
232935  6     31.687542       1.166188e-05   
232657  1     30.694159       2.381163e-06   
234793  19    29.244334       2.072846e-02   
234794  20    28.060164       3.805248e-03   
234796  22    27.656797       9.937782e-04   
234797  23    27.292580       2.068391e-02   
232936  7     27.113830       6.148038e-05   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               

In [22]:
shorten_names = {
                 'cellular response to DNA damage stimulus': 'DDR',
                 'negative regulation of apoptotic process': 'negative regulation of apoptosis',
                 'Apoptosis_Homo sapiens_R-HSA-109581' : 'Apoptosis',
                }
renamed = time_24_hour_prot.copy()
time_24_hour_prot['term_name'] = time_24_hour_prot['term_name'].replace(shorten_names)
term_net_24, mol_net_24 = nt.create_subnetwork(shorten_names.values(), time_24_hour_prot, network, '24hr', cytoscape_js=False)

('DDR', 20)
('negative regulation of apoptosis', 27)
('Apoptosis', 26)
Looking for direct edges


In [23]:
view.display_graph(term_net_24)

In [24]:
view.display_graph(mol_net_24, add_parent=True )

In [25]:
time_48_hour_prot = et.filter_dataframe(proteomics, p_value=0.05, combined_score=0.0, category='proteomics_up', sample_id='48hr', db=process_dbs)
slimmed = et.filter_similar_terms(time_48_hour_prot, threshold=.5)
display(slimmed.head(25))

Number of rows went from 419 to 167


term_name  \
258340  mRNA splicing, via spliceosome                                                                  
258341  neutrophil degranulation                                                                        
257060  Gene Expression_Homo sapiens_R-HSA-74160                                                        
258342  mitochondrial ATP synthesis coupled proton transport                                            
257062  Metabolism of proteins_Homo sapiens_R-HSA-392499                                                
256490  mRNA processing_Mus musculus_WP310                                                              
258344  membrane organization                                                                           
258346  mRNA export from nucleus                                                                        
258347  negative regulation of apoptotic process                                                        
258348  RNA metabolic process                                                                           
258349  platelet degranulation                                                                          
257063  Disease_Homo sapiens_R-HSA-1643685                                                              
256492  Electron Transport Chain_Homo sapiens_WP111                                                     
258350  ER to Golgi vesicle-mediated transport                                                          
257065  Hemostasis_Homo sapiens_R-HSA-109582                                                            
256493  XPodNet - protein-protein interactions in the podocyte expanded by STRING_Mus musculus_WP2309   
257066  Vesicle-mediated transport_Homo sapiens_R-HSA-5653656                                           
255961  Protein processing in endoplasmic reticulum_Homo sapiens_hsa04141                               
255962  Focal adhesion_Homo sapiens_hsa04510                                                            
258352  RNA splicing                                                                                    
256206  ErbB1 downstream signaling_Homo sapiens_30d60550-6192-11e5-8ac5-06603eb7f303                    
258353  regulation of cellular response to heat                                                         
258354  rRNA processing                                                                                 
256496  Integrated Pancreatic Cancer Pathway_Homo sapiens_WP2377                                        
258355  cellular protein metabolic process                                                              

        rank  combined_score   adj_p_value  \
258340  1     234.079702      6.840193e-24   
258341  2     180.858577      5.282489e-13   
257060  3     92.138380       4.733666e-17   
258342  3     90.680332       2.328503e-12   
257062  5     82.654616       2.497306e-15   
256490  1     81.175763       3.223967e-15   
258344  5     75.662017       7.658452e-07   
258346  7     67.106534       4.281073e-09   
258347  8     61.614403       7.827425e-05   
258348  9     57.796749       2.042800e-09   
258349  10    57.718950       2.573875e-07   
257063  6     56.883870       2.740126e-09   
256492  3     55.883408       6.727001e-11   
258350  11    52.615159       3.217147e-06   
257065  8     52.388225       1.783685e-09   
256493  4     52.009261       2.495023e-09   
257066  9     50.489896       3.569055e-09   
255961  2     47.316584       4.097873e-10   
255962  3     46.294756       1.486071e-09   
258352  13    44.601721       8.424171e-06   
256206  1     39.575858       5.383828e-08   
258353  14    35.933418       6.080024e-05   
258354  15    35.377604       5.070748e-04   
256496  7     35.351683       1.108409e-06   
258355  16    35.341205       7.286813e-04   

                                                                                                                                                                    

In [29]:
shorten_names = {
                 'membrane organization': 'Membrane Organization',
                 'negative regulation of apoptotic process': 'negative regulation of apoptosis',
                 'neutrophil degranulation' : 'neutrophil degranulation',
                }
time_48_hour_prot['term_name'] = time_48_hour_prot['term_name'].replace(shorten_names)
term_net_48, mol_net_48 = nt.create_subnetwork(shorten_names.values(), time_48_hour_prot, network, '48hr', cytoscape_js=False)

('Membrane Organization', 31)
('negative regulation of apoptosis', 34)
('neutrophil degranulation', 63)
Looking for direct edges


In [30]:
view.display_graph(term_net_48)

In [33]:
view.render_graph(mol_net_48)